zcm/julia/ZCM.jl

```diff
@@ -0,0 +1,334 @@
+module ZCM
+
+# RRR / Note: Julia requires that the memory layout of the C structs is consistent
+#             between their definitions in zcm headers and this file
+
+# C ptr types for types that we don't need the internals of
+module Native
+
+type Zcm
+end
+
+type Sub
+end
+
+type UvSub
+end
+
+type RecvBuf
+    recv_utime ::Int64;
+    zcm        ::Ptr{Native.Zcm};
+    data       ::Ptr{UInt8};
+    data_size  ::UInt32;
+end
+
+type EventLog
+end
+
+type EventLogEvent
+    eventnum  ::Int64;
+    timestamp ::Int64;
+    chanlen   ::Int32;
+    datalen   ::Int32;
+    channel   ::Ptr{UInt8};
+    data      ::Ptr{UInt8};
+end
+
+end
+
+# RRR: make variable naming consistent
+# RRR: improve orgainzation and make header comments more visible
+# Exported Objects and Methods
+type RecvBuf
+    rbuf  ::Ptr{Native.RecvBuf};
+    utime ::Int64;
+    data  ::Array{UInt8};
```

Comment:
This should be `Array{UInt8, 1}` or (equivalently, because it's just an alias), `Vector{UInt8}`. Currently, this field can hold an array with any number of dimensions, but I'm pretty sure you just have 1-D vectors. Changing the field will make its type concrete and improve performance.